moneybird.js

```diff
@@ -1,21 +1,124 @@
 const SuperAgent = require('superagent');
 const {MONEYBIRD_API_TOKEN, MONEYBIRD_ADMINISTRATION_ID} = process.env;
+const {keyBy} = require('lodash');
+const {ENV} = process.env;
+const Fetch = require('./fetch.js');
+const DevFetch = require('./devFetch.js');
+const yaml = require('js-yaml');
 
-exports.getContactData = async (contactId) => {
+const use_cache = false;
 
-    const contact_request = await SuperAgent
-        .get(`https://moneybird.com/api/v2/${MONEYBIRD_ADMINISTRATION_ID}/contacts/${contactId}`)
-        .set('Authorization', `Bearer ${MONEYBIRD_API_TOKEN}`)
-        .set('Accept', 'application/json')
-    return contact_request.body
+var fetch;
+switch (ENV) {
+    case 'prod':
+    case 'test':
+        fetch = new Fetch();
+        break;
+    case 'dev':
+    default:
+        fetch = new DevFetch();
+        break;
 }
 
-exports.getInvoices = async (contactId, reference = null) => {
+const loadSubscriptionFile = async (req) =>
+    yaml.load(await fetch.fetchFromFile(req, 'subscription.yml'))
 
-    const invoices_request = await SuperAgent
-        .get(`https://moneybird.com/api/v2/${MONEYBIRD_ADMINISTRATION_ID}/sales_invoices?filter=contact_id:${contactId}`)
-        .set('Authorization', `Bearer ${MONEYBIRD_API_TOKEN}`)
-        .set('Accept', 'application/json')
+// list of invoices as received from moneybird, filtered by contact_id
+var moneybird_invoices_cache = {};
+// the above, but indexed by invoice.id for easy lookup
+var moneybird_invoices_index = {};
+// contact data as received from moneybird
+var moneybird_contact_data_cache = {};
 
-    return invoices_request.body.filter( invoice => invoice.reference == reference )
+const getProjectName = (req) => {
+    if (ENV === 'dev') {
+        return '_development';
```

Comment:
Same comment about test variations in production code. Do we really need this?

Reply:
While running in development mode, requests do not include a project name. 
This is a workaround, to make the code work even when no project name is available.

A better solution might be to always include a project name.

Reply:
something like "hyperdev-development", to prevent it from clashing with our clients